controller/array_action/array_mediator_svc.py

```diff
@@ -846,6 +846,35 @@ def _get_detailed_hosts_list_cmd(self, host_list):
             writer.write(LIST_HOSTS_CMD_FORMAT.format(HOST_ID=host.id))
         return writer.getvalue()
 
+    def _lshost(self, **kwargs):
+        cli_host = self.client.svcinfo.lshost(**kwargs).as_single_element
+        if not cli_host:
+            raise array_errors.HostNotFoundError(kwargs)
+        return cli_host
+
+    def _get_cli_host_by_id(self, host_id):
+        return self._lshost(object_id=host_id)
+
+    def _get_cli_host_by_name(self, host_name):
+        filter_value = 'name={}'.format(host_name)
+        return self._lshost(filtervalue=filter_value)
+
+    def get_host_by_name(self, host_name):
+        cli_host_by_name = self._get_cli_host_by_name(host_name)
+        cli_host_by_id = self._get_cli_host_by_id(cli_host_by_name.id)
```

Comment:
```suggestion
    def _get_cli_host(self, id_or_name):
        cli_host = self.client.svcinfo.lshost(object_id=id_or_name).as_single_element
        if not cli_host:
            raise array_errors.HostNotFoundError(kwargs)
        return cli_host

    def get_host_by_name(self, host_name):
        cli_host = self._get_cli_host(host_name)
```
sorry for the confusion

Reply:
so never actually needed the `filtervalue` for `name`?

Reply:
exactly

Reply:
done